File: tbsmith/day3.php.com/views/helpers/form_builder.php
<?php

Class FormBuilder {

    private $validation_data;
    private $form_data;

    function __construct($validation_data = array(), $form_data = array()) {
        $this->validation_data = $validation_data;
        $this->form_data = $form_data;
    }

    function formHeader($action) {
        echo '<form action="'.$action.'" '.
             'enctype="multipart/form-data"
             method="post">' . "\n";
    }

    function formCloser() {
        echo "</form>\n";
    }

    function errorRow($name) {
        echo '<div class="error_row">' . $this->validation_data[$name]['error_message'] . "</div>\n";
    }

    function textInput($name, $password=false) {
        $type = $password ? 'password' : 'text';
        if (!empty($this->validation_data[$name]) && $this->validation_data[$name]['has_error'] == true) {
            $this->errorRow($name);
        }

        if (!empty($this->form_data[$name])) {
            $current_value = $this->form_data[$name];
        } else {
            $current_value = '';
        }

        echo "<div class=\"input_row\">\n" . 
            "<div class=\"input_left\">\n" .
            ucfirst($name) . 
            '</div>' .
            "<div class=\"input_right\">\n" .
            "<input name=\"$name\" type=\"$type\" value=\"" .
                $current_value . "\">" .
            "</div>\n".
            "</div>\n";
    }

    function radio($name, $options) {
        if (!empty($this->validation_data[$name]) && $this->validation_data[$name]['has_error'] == true) {
            $this->errorRow($name);
        }       
        echo "<div class=\"input_row\">\n";
        echo "<div class=\"input_left\">\n";
        echo ucfirst($name) . "\n";
        echo "</div>\n";
        echo "<div class=\"input_right\">\n";
        foreach ($options as $option) {
            if (!empty($this->form_data[$name]) && $this->form_data[$name] == $option) {
                $checked = 'checked';
            } else {
                $checked = '';
            }
            echo $option . "<input name=\"$name\" type=\"radio\" $checked value=\"$option\"/>\n";
        }
        echo "</div>\n";
        echo "</div>\n";
    }

    function submit() {
        echo "<div class=\"input_row\">\n";
        echo "<input type=\"submit\">\n";
        echo "</div>\n";
    }

}

?>

File: tbsmith/day3.python.com/views/helpers/form_builder.py
#!/usr/bin/python


class FormBuilder():

    def __init__(self, validation_data={}, form_data={}):
        self.validation_data = validation_data
        self.form_data = form_data

    def formHeader(self, action):
        print "<form action=\"" + action + "\" "
        print "enctype=\"multipart/form-data\" "
        print "method=\"post\">\n"

    def formCloser(self):
        print "</form>\n"

    def errorRow(self, name):
        print '<div class="error_row">'
        print self.validation_data[name]['error_message']
        print "</div>\n"

    def textInput(self, name, password=False):
        if password is True:
            type = 'password'
        else:
            type = 'text'

        if name in self.validation_data and \
           self.validation_data[name] and \
           self.validation_data[name]['has_error'] is True:
            self.errorRow(name)

        if name in self.form_data:
            current_value = self.form_data.getvalue(name)
        else:
            current_value = ""

        print "<div class=\"input_row\">\n"
        print "<div class=\"input_left\">\n"
        print name
        print "</div>"
        print "<div class=\"input_right\">\n"
        print "<input name=\"" + name + "\" "
        print "type=\"" + type + "\" value=\""
        print current_value + "\">"
        print "</div>\n"
        print "</div>\n"

    def radio(self, name, options):
        if name in self.validation_data and \
           self.validation_data[name] and \
           self.validation_data[name]['has_error'] is True:
            self.errorRow(name)
        print "<div class=\"input_row\">\n"
        print "<div class=\"input_left\">\n"
        print name
        print "</div>"
        print "<div class=\"input_right\">\n"
        for option in options:
            if name in self.form_data and \
               self.form_data.getvalue(name) == option:
                checked = 'checked'
            else:
                checked = ''
            print option
            print "<input name=\"" + name + "\" "
            print "type=\"radio\" " + checked + " "
            print "value=\"" + option + "\"/>\n"
        print "</div>\n"
        print "</div>\n"

    def submit(self):
        print "<div class=\"input_row\">\n"
        print "<input type=\"submit\">\n"
        print "</div>\n"
